### DISCUSSION TITLE
Save an Error Log automatically when a Flow crashes

### DISCUSSION THREAD

**Discussion by fabm97** [jongpie/NebulaLogger#261] [Ideas]
Hello,

we are using Nebula Logger to keep track of Flow Errors and historize some outcomes of the Flows for future reference.
Currently when we want to log a crash of a flow, we need to implement a fault path on a specific Element of the Flow beforehand so a Log will be created if the Flow crashes. 
![image](https://user-images.githubusercontent.com/40338881/148946199-11c71610-2c7f-4fbe-b83c-08447a611518.png)

We have a complex flow structure with many big flows implementing Subflows in our org and want to Log any crashes so that we can track them in a Dashboard. 
We came to the idea that it would be awesome to have a possibility to create Flow Logs automatically for certain Flows regardless of why and where they crash. The basic Idea would be to somehow automatically create a log with the flow error message when a Flow crashes.
This would enable users of Nebula Logger to more easily use Salesforce Reporting to report on Flow Errors and historize them.
What do you think of this idea and do you think this is possible to implement?

**Comment by jongpie** [comment]
Hi @fabm97, thanks for the feedback! I do think that this a fantastic idea - right now, it's a bit of a pain to properly setup complex Flows to have all of the necessary error logging & handling. I checked internally at Salesforce, and unfortunately, I don't think that this is currently possible due to how the platform currently works - there is not currently a way to setup a single fault path for an entire Flow, so we are unfortunately stuck with the current approach of adding Flow fault paths for every element that could possibly fail. 

If/when Salesforce every provides a more streamlined way to handle errors in Flows, I will definitely see if I can improve Nebula Logger's approach to make it easier to log Flow errors.
